main.go

```diff
@@ -90,6 +92,17 @@ func main() {
 	}
 }
 
+func readPassword(prompt string) (string, error) {
+	state, err := terminal.MakeRaw(int(os.Stdin.Fd()))
+	if err != nil {
+		return "", err
+	}
+	defer terminal.Restore(int(os.Stdin.Fd()), state)
```

Comment:
`/dev/tty` represents the controlling terminal for the process. If you use that instead of stdin, it should work with pipes too. Should be available on on all POSIX systems afaik.